dali/pipeline/workspace/sample_workspace.h

```diff
@@ -134,6 +134,17 @@ class DLL_PUBLIC SampleWorkspace : public WorkspaceBase<SampleInputType, SampleO
 DLL_PUBLIC void MakeSampleView(SampleWorkspace& sample, HostWorkspace& batch, int data_idx,
                                  int thread_idx);
 
+/**
+ * @brief After running sample-wise operator we need to fix the Tensor Vector guarantees that were
+ * broken by the legacy operators operating just on samples. We propagate the properties from
+ * samples and enforce they are consistent.
+ *
+ * TODO(klecki): Introduce RAII wrapper for MakeSampleView and EnforceCorrectness
+ * @param batch The workspace to update after executing samplewise operator
+ * @param contiguous If the operator infers outputs and thus uses contiguous allocations
+ */
+DLL_PUBLIC void EnforceCorrectness(HostWorkspace& ws, bool contiguous);
```

Comment:
This name is non-descriptive. Does it _enforce_ or _propagate_? What is propagated? What is the result of this function?

Reply:
I tried my best to fix it.